project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+
+Examples will be included in the rendered output in the order given here: so for example you may want to present the basic example first in the array, then more complex ones.
+
+Each example is specified as a directory containing four files, one mandatory and three optional:
+
+-   "description.md" - the only mandatory one
+-   "example.html"
+-   "example.css"
+-   "example.js"
+
+**Note: we don't yet support "hidden" sources One suggestion was just to call them "hidden.css" etc.**
+
+#### description.md
+
+A Markdown file containing some YAML front matter.
+
+The front matter contains a mandatory `title` and an optional `height`:
+
+```
+---
+title: Custom error messages
+height: 192
+---
+```
+
+If `height` is included, then stumptown treats this example as a live sample, and `height` is the height of the "Result" iframe in pixels (the iframe always gets full page width).
+
+If `height` is omitted, then stumptown treats this example as just a static example.
+
+**Note: many examples currently omit "title".**
+
+**Note: we have also considered having a less implicit way to signal whether or not an example is live (https://github.com/mdn/stumptown-content/issues/45 ).**
+
+The rest of the "description.md" file is a prose description of the example.
+
+#### example.\*
+
+The example.html, example.css, and example.js files contain the example sources themselves.
+
+### specifications
+
+This property lists specifications for the item.
+
+This is given as one of:
+
+-   the string "non-standard"
+-   a single URL
+-   an array of URLs
+
+URLs must match one of the URLs listed in "specifications.yaml" (see the "Specifications" section), where "match" means they fully contain it, although they may add additional path or hash items. For example:
+
+    https://html.spec.whatwg.org/multipage/embedded-content.html#the-audio-element
+
+matches an entry like:
+
+    html.spec.whatwg.org/multipage
+
+### info_box
+
+TBD
+
+### browser_compatibility
+
+A string representing a BCD query, e.g. "html.elements.abbr".
+
+### link_lists
+
+An object used to describe a list of lists of internal links. This is currently used only in landing pages and sidebars, but might be useful in other contexts as well.
+
+This is specified as an array of "link list" objects. Each of these objects will have a title and a list of links. Each link has a URL, a title, and a short description. This enables the renderer to build landing pages like:
+
+```
+## Title of a list of Links
+
+* [link text 1](link target 1): short description for the link 1
+* [link text 2](link target 2): short description for the link 2
+...
+
+## Title of another list of Links
+
+* [link text 3](link target 3): short description for the link 3
+* [link text 4](link target 4): short description for the link 4
+...
+```
+
+**Note: we don't currently support short descriptions for guide pages. But we should.**
```

Comment:
For the single page types of link list objects, we should allow the addition of an optional "description" property to override the description that would otherwise be fetched from the target page.

Reply:
I don't think we should do this, without a much better reason. I don't want to start embedding markup in YAML, and I think it would get hard to maintain.